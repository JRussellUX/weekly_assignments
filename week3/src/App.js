import { useState } from "react";
import Quote from "./Quote";
import Header from "./Header";

//function getRandomQuote(quotes) {
//  return quotes[Math.floor(Math.random() * quotes.length)];
//}

function App() {
  const [quote, setQuote] = useState(null);

  function getNewQuote() {
    fetch("http://thesimpsonsquoteapi.glitch.me/quotes")
      .then((res) => res.json())
      .then((json) => {
        setQuote(json[0]);
      });
  }

  return (
    <main>
      <Header text="Week3 Assignment: Simpsons Quote Generator" />
      <section>
        <button onClick={getNewQuote}>New Quote</button>
        <Quote quote={quote}></Quote>
      </section>
    </main>
  );
}

export default App;
