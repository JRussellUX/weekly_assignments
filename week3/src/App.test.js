import renderer from 'react-test-renderer';
import App from './App';
import {render, screen} from '@testing-library/react';

test('renders quote screen', () => {
  const app = renderer.create(<App />);
  let tree = app.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Button appears', () => {
  render(<App />);
  const button = screen.getByText("New Quote");
  expect(button).toBeTruthy();
});
