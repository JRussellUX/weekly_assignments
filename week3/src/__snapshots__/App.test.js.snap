// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders quote screen 1`] = `
<main>
  <h1>
    Week3 Assignment: Simpsons Quote Generator
  </h1>
  <section>
    <button
      onClick={[Function]}
    >
      New Quote
    </button>
  </section>
</main>
`;
